### DISCUSSION TITLE
Option to not decrypt some https requests

### DISCUSSION THREAD

**Discussion by andn99** [mitmproxy/mitmproxy#6118] [Q&A]
Hi guys, is there any option to not decrypt some https requests, such as requests to banking websites. I still want the proxy to intercept those requests so that my script can see and track those requests, but don't want the proxy to decrypt them because of sensitive data.
At the moment I can use ignore_hosts to not intercept those requests, but in that case my addon script can't see them either. 
Thank you!

**Comment by mhils** [comment]
If you don't want them to be decrypted, you can either use [`ignore_hosts` or `tcp_hosts`](https://docs.mitmproxy.org/stable/concepts-options/). The former will not appear in any event hooks, the latter will make the encrypted traffic appear in the [`tcp_*` events](https://docs.mitmproxy.org/stable/api/events.html#TCPEvents).

**Reply by andn99** [reply]
Hi @mhils , I started my proxy from the script below. However, none of `tcp_*` events in my TCPFilter was triggered when I visited `reqres.in` or `example.com`. The requests are still go through my http events in my RequestFilter. Did I miss any configuration? Thanks in advance!

```
async def start_proxy(host, port):
    opts = options.Options(
        listen_host=host, listen_port=port, tcp_hosts=["reqres.in", "example.com"]
    )

    master = dump.DumpMaster(
        opts,
        with_termlog=False,
        with_dumper=False,
    )
    master.addons.add(TCPFilter())
    master.addons.add(RequestFilter())
    master.addons.add(asgiapp.ASGIApp(app, API_ADDON_HOST, API_ADDON_PORT))
    await master.run()
    return master
```

**Reply by mhils** [reply]
See https://stackoverflow.com/questions/68654077/is-there-a-way-to-start-mitmproxy-v-7-0-2-programmatically-in-the-background/68661172#68661172 - manually instantiating DumpMaster is at your own risk. :)

**Comment by andn99** [comment]
Thanks @mhils ! I'll try with the tcp_hosts.
